components/tidb_query_vec_executors/src/index_scan_executor.rs

```diff
@@ -232,27 +245,110 @@ impl IndexScanExecutorImpl {
                 return Err(other_err!("Unexpected missing column {}", col_id));
             }
         }
+        Ok(())
+    }
+
+    fn extract_pk_columns_from_datum_payload(
```

Comment:
maybe call it `extract_columns_from_datums_format`, since you are calling it even for old index as well (where they are just normal index columns instead of PK columns)

Reply:
Done